pkg/receive/writecapnp/write_request.go

```diff
@@ -59,7 +61,9 @@ func NewRequest(wr WriteRequest) (*Request, error) {
 	start := uint32(0)
 	for i := 0; i < offsets.Len(); i++ {
 		end := offsets.At(i)
-		*strings = append(*strings, string(data[start:end]))
+
+		b := data[start:end]
+		*strings = append(*strings, unsafe.String(&b[0], len(b)))
```

Comment:
Nice. Do you think this could prevent the byte slice from the write request from getting GC'd?

Reply:
Actually now that I think about this, we copy the labelset if it's not found using the series ref. So the string from the request is never retained.

Reply:
This prevents from making a copy; I think string([]byte) always allocates because strings are immutable in Go